src/app/clusters/commodity-tariff-server/commodity-tariff-server.h

```diff
@@ -0,0 +1,172 @@
+    X(TariffUnit,                   DataModel::Nullable<Globals::TariffUnitEnum>) \
+    X(StartDate,                    DataModel::Nullable<uint32_t>) \
+    X(DefaultRandomizationOffset,   DataModel::Nullable<int16_t>) \
+    X(DefaultRandomizationType,     DataModel::Nullable<DayEntryRandomizationTypeEnum>)
+    /*X(CalendarPeriods,              DataModel::List<Structs::CalendarPeriodStruct::Type>) \
+    X(DayPatterns,                  DataModel::List<Structs::DayPatternStruct::Type>) \
+    X(IndividualDays,               DataModel::List<Structs::DayStruct::Type>) \
+    X(DayEntries,                 DataModel::List<Structs::DayEntryStruct::Type>) \
+    X(TariffPeriods,                DataModel::List<Structs::TariffPeriodStruct::Type>) \
+    X(TariffComponents,             DataModel::List<Structs::TariffComponentStruct::Type>)*/
+
+#define COMMODITY_TARIFF_PRIMARY_ATTRIBUTES_DEV \
+    X(TariffInfo,                   DataModel::Nullable<Structs::TariffInformationStruct::Type>) \
+    X(TariffPeriods,                DataModel::List<Structs::TariffPeriodStruct::Type>) \
+    X(DayEntries,                   DataModel::List<Structs::DayEntryStruct::Type>) \
+    X(TariffComponents,             DataModel::List<Structs::TariffComponentStruct::Type>) \
+    X(IndividualDays,               DataModel::List<Structs::DayStruct::Type>) \
+    X(DayPatterns,                  DataModel::List<Structs::DayPatternStruct::Type>) \
+    X(CalendarPeriods,              DataModel::List<Structs::CalendarPeriodStruct::Type>)
+
+#define COMMODITY_TARIFF_PRIMARY_ATTRIBUTES \
+    COMMODITY_TARIFF_PRIMARY_ATTRIBUTES_STUBS \
+    COMMODITY_TARIFF_PRIMARY_ATTRIBUTES_DEV
+
+// Centralized attribute declaration (Current attrs)
+#define COMMODITY_TARIFF_CURRENT_ATTRIBUTES \
+    X(CurrentDay,                   DataModel::Nullable<Structs::DayStruct::Type>) \
+    X(NextDay,                      DataModel::Nullable<Structs::DayStruct::Type>) \
+    X(CurrentDayEntry,              DataModel::Nullable<Structs::DayEntryStruct::Type>) \
+    X(NextDayEntry,                 DataModel::Nullable<Structs::DayEntryStruct::Type>) \
+    X(CurrentDayEntryDate,          DataModel::Nullable<uint32_t>) \
+    X(NextDayEntryDate,             DataModel::Nullable<uint32_t>) \
+    X(CurrentTariffComponents,      DataModel::List<Structs::TariffComponentStruct::Type>) \
+    X(NextTariffComponents,         DataModel::List<Structs::TariffComponentStruct::Type>)
+
+class Delegate
+{
+public:
+    Delegate()          = default;
+    virtual ~Delegate() = default;
+
+    void SetEndpointId(EndpointId aEndpoint) { mEndpointId = aEndpoint; }
+
+
+    virtual Protocols::InteractionModel::Status GetDayEntryById(DataModel::Nullable<uint32_t> aDayEntryId,
+                                                                Structs::DayEntryStruct::Type & aDayEntry) = 0;
+
+    virtual Protocols::InteractionModel::Status GetTariffComponentInfoById(DataModel::Nullable<uint32_t>  aTariffComponentId,
```

Comment:
What is the lifetime of the buffers the outparams point to?  Needs to be documented at the very least.